source/microbit/microbitdisplay.cpp

```diff
@@ -432,7 +439,13 @@ mp_obj_t microbit_display_scroll_func(mp_uint_t n_args, const mp_obj_t *pos_args
     mp_arg_val_t args[MP_ARRAY_SIZE(scroll_allowed_args)];
     mp_arg_parse_all(n_args - 1, pos_args + 1, kw_args, MP_ARRAY_SIZE(scroll_allowed_args), scroll_allowed_args, args);
     mp_uint_t len;
-    const char* str = mp_obj_str_get_data(args[0].u_obj, &len);
+    mp_obj_t object_string;
+    if (!MP_OBJ_IS_TYPE(args[0].u_obj, &mp_type_str)) {
```

Comment:
I think that scroll should behave the same as show here, in the sense that it should only accept ints, floats and strings (not arbitrary objects).  Converting objects like lists, tuples and dicts to a string to show them is arguably not what you want as the programmer, and conflicts with show which would take a list/tuple and show each element of the list/tuple.